### ISSUE TITLE
Error by initialize with Podman: `sudo: no tty present and no askpass program specified`

### ISSUE THREAD

**Issue by lroehrs** [dockware/dockware#42]
We have tried to launch Dockware via Podman at Profihost.

After starting the container the following error occurs:

```
....
*******************************************************
** DOCKWARE IMAGE: dev
** Tag: 6.3.5.0
** Version: 1.3.5-beta
** Built: Mon Feb  1 14:32:36 UTC 2021
** Copyright 2020 dasistweb GmbH
*******************************************************

launching dockware...please wait...

DOCKWARE: setting timezone to Europe/Berlin...
sudo: no tty present and no askpass program specified
$ 
```

After adding the user "www-data" to the sudo group (`usermod -aG sudo www-data`) the container is started:
```
....
WOHOOO, dockware/dev:6.3.5.0 IS READY :) - let's get started
-----------------------------------------------------
....
```

I assume that if the user `www-data` is added to the `sudo` group in the `dockerfile` it will work directly. 
I can't estimate if this has unintended side effects.

Let me know, if you need something else from me.

**Comment by boxblinkracer** [comment]
hi there

i normally get this only if i do execute something from outside
so like a deployment pipeline where the user (deploy ssh user) has no tty
then it would work with > docker exec my-container bash -C "do something"
thats the non-tty option

a startup should not be involved in anyhow....

can you send me your pod setup and what you use to start the container?

i really try to avoid www-data being added to the sudo, especially not with no-pwd :) that would be a problem.
it should actually start with a user "dockware" that has sudo,
could it be that you try to start it with user www-data instead? then it totally makes sense

thank you

**Comment by lroehrs** [comment]
Hi, 

thanks for the fast reply.
The startup is with ´www-data`.
If this is already the wrong user, it makes sense, yes.

Startup: 

```
podman@cloud2-vm400:~$ podman system reset

WARNING! This will remove:
        - all containers
        - all pods
        - all images
        - all build cache
Are you sure you want to continue? [y/N] y
podman@cloud2-vm400:~$ podman run -p 2080:80 dockware/dev:latest
Trying to pull docker.io/dockware/dev:latest...
Getting image source signatures
Copying blob d519e2592276 done  
....
Copying config 596684f203 done  
Writing manifest to image destination
Storing signatures

 _____   ____   _____ _  ____          __     _____  ______ 
|  __ \ / __ \ / ____| |/ /\ \        / /\   |  __ \|  ____|
| |  | | |  | | |    | ' /  \ \  /\  / /  \  | |__) | |__   
| |  | | |  | | |    |  <    \ \/  \/ / /\ \ |  _  /|  __|  
| |__| | |__| | |____| . \    \  /\  / ____ \| | \ \| |____ 
|_____/ \____/ \_____|_|\_\    \/  \/_/    \_\_|  \_\______|

68 69 20 64 65 76 65 6C 6F 70 65 72 2C 20 6E 69 63 65 20 74 6F 20 6D 65 65 74 20 79 6F 75
6c 6f 6f 6b 69 6e 67 20 66 6f 72 20 61 20 6a 6f 62 3f 20 77 72 69 74 65 20 75 73 20 61 74 20 6a 6f 62 73 40 64 61 73 69 73 74 77 65 62 2e 64 65

*******************************************************
** DOCKWARE IMAGE: dev
** Tag: 6.3.5.0
** Version: 1.3.5-beta
** Built: Mon Feb  1 14:32:36 UTC 2021
** Copyright 2020 dasistweb GmbH
*******************************************************

launching dockware...please wait...

DOCKWARE: setting timezone to Europe/Berlin...
sudo: no tty present and no askpass program specified
podman@cloud2-vm400:~$
```

Image Details: 
```
podman@cloud2-vm400:~$ podman inspect 982b3d165885
[
    {
        "Id": "982b3d16588567bd5a7901c875a12cb52f5bc1537081cc9a98786bfb3e2d0925",
        "Created": "2021-02-08T16:25:54.8045019+01:00",
        "Path": "/entrypoint.sh",
        "Args": [
            "/entrypoint.sh"
        ],
        "State": {
            "OciVersion": "1.0.2-dev",
            "Status": "exited",
            "Running": false,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 0,
            "ExitCode": 1,
            "Error": "",
            "StartedAt": "2021-02-08T16:25:54.936398637+01:00",
            "FinishedAt": "2021-02-08T16:25:55.052692897+01:00",
            "Healthcheck": {
                "Status": "",
                "FailingStreak": 0,
                "Log": null
            }
        },
        "Image": "596684f203db4ed337ac0d4705f8fe43a89ca359f0e49d8491a3ceee6faad8a5",
        "ImageName": "docker.io/dockware/dev:latest",
        "Rootfs": "",
        "Pod": "",
        "ResolvConfPath": "/run/user/1000/containers/overlay-containers/982b3d16588567bd5a7901c875a12cb52f5bc1537081cc9a98786bfb3e2d0925/userdata/resolv.conf",
        "HostnamePath": "/run/user/1000/containers/overlay-containers/982b3d16588567bd5a7901c875a12cb52f5bc1537081cc9a98786bfb3e2d0925/userdata/hostname",
        "HostsPath": "/run/user/1000/containers/overlay-containers/982b3d16588567bd5a7901c875a12cb52f5bc1537081cc9a98786bfb3e2d0925/userdata/hosts",
        "StaticDir": "/home/podman/.local/share/containers/storage/overlay-containers/982b3d16588567bd5a7901c875a12cb52f5bc1537081cc9a98786bfb3e2d0925/userdata",
        "OCIConfigPath": "/home/podman/.local/share/containers/storage/overlay-containers/982b3d16588567bd5a7901c875a12cb52f5bc1537081cc9a98786bfb3e2d0925/userdata/config.json",
        "OCIRuntime": "crun",
        "LogPath": "/home/podman/.local/share/containers/storage/overlay-containers/982b3d16588567bd5a7901c875a12cb52f5bc1537081cc9a98786bfb3e2d0925/userdata/ctr.log",
        "LogTag": "",
        "ConmonPidFile": "/run/user/1000/containers/overlay-containers/982b3d16588567bd5a7901c875a12cb52f5bc1537081cc9a98786bfb3e2d0925/userdata/conmon.pid",
        "Name": "quizzical_poincare",
        "RestartCount": 0,
        "Driver": "overlay",
        "MountLabel": "",
        "ProcessLabel": "",
        "AppArmorProfile": "",
        "EffectiveCaps": null,
        "BoundingCaps": [
            "CAP_AUDIT_WRITE",
            "CAP_CHOWN",
            "CAP_DAC_OVERRIDE",
            "CAP_FOWNER",
            "CAP_FSETID",
            "CAP_KILL",
            "CAP_MKNOD",
            "CAP_NET_BIND_SERVICE",
            "CAP_NET_RAW",
            "CAP_SETFCAP",
            "CAP_SETGID",
            "CAP_SETPCAP",
            "CAP_SETUID",
            "CAP_SYS_CHROOT"
        ],
        "ExecIDs": [],
        "GraphDriver": {
            "Name": "overlay",
            "Data": {
                "LowerDir": "/home/podman/.local/share/containers/storage/overlay/639999a6ce086af995bcca672dde27460f28a29074aa6b3d43915043a6943203/diff:/home/podman/.local/share/containers/storage/overlay/0dfbc65927b3a3db3c87422bdbb9496c2ca695c6c26aeeeac0e5f4acfd429ba6/diff:/home/podman/.local/share/containers/storage/overlay/b15b2a9240ee93de2768d1d302d7b3731591905c64c094d3083f1180faa31a93/diff:/home/podman/.local/share/containers/storage/overlay/099ac4a5585fb3b6ee027a102901f6ce2528efe6b83b6bef0ab1d83dbb154f0b/diff:/home/podman/.local/share/containers/storage/overlay/84120d4a4a2b8d069caf1988bfd29d597cec68289394187d1965f115c81e5450/diff:/home/podman/.local/share/containers/storage/overlay/b08c086bfba9c7ac88dc86dd05423a306b496fca56f493bfae6d2a4d7d311b76/diff:/home/podman/.local/share/containers/storage/overlay/ed023991751fa14b388670310adad206e94b80bff84b02c9f06864c228b91ef1/diff:/home/podman/.local/share/containers/storage/overlay/b845c88d6bc23ca1b4360d61b3f8560ba8952db3ed7654c824609b2a887e919e/diff:/home/podman/.local/share/containers/storage/overlay/6e2999b5c17389afc639b8c94726210265c67c8d5b74c0c56355ed00e9769a9b/diff:/home/podman/.local/share/containers/storage/overlay/0513974c8976e39b51456aaedd26c1d4d870d0598d325b55ccf8984f1334542f/diff:/home/podman/.local/share/containers/storage/overlay/8e6ec6f380ca2e98672079ca04a85c6baf7984b95bd266b42681f2422064a237/diff:/home/podman/.local/share/containers/storage/overlay/5c27ea5cab0e1710abdb1b96e52bebfee745b1ea1cd72cec05f0d15a12be6f07/diff:/home/podman/.local/share/containers/storage/overlay/19459c6d94b5c14fac5b0bb1d005016d4831df924b4ee8bc2399a1465d47738d/diff:/home/podman/.local/share/containers/storage/overlay/53d3f719555afe519038b971da999f14919cd437af31bc92c7af88f367f1b529/diff:/home/podman/.local/share/containers/storage/overlay/f86e0d1ceb1a5329b5dacb0df9aeec74676cf0732335274ca063e4d1e9bb3b06/diff:/home/podman/.local/share/containers/storage/overlay/dcda8b5a3c930023bae46b78f43f996c05cc86e4527396f1b680a201bd74b45b/diff:/home/podman/.local/share/containers/storage/overlay/b08491fb02a309f0e63fb050864985301d5af31f374b99ebd236fd6f017b0b05/diff:/home/podman/.local/share/containers/storage/overlay/fb9096c47ea6a122d1ddccd38b4e153c1c9593d5092e5b9948d04f1097148d74/diff:/home/podman/.local/share/containers/storage/overlay/f56bebd226fda008d489ee3cfb5eaff310a81ae54ae84005ad97a3bd1b52c21e/diff:/home/podman/.local/share/containers/storage/overlay/cba108b5f22b4974e470d363c934c2708eb42da11b8696ba637a06eaac2c3644/diff:/home/podman/.local/share/containers/storage/overlay/eba926d225472e90b17ab7edc4b67ec8691aeb026bb885d1457387a03ac7e6fb/diff:/home/podman/.local/share/containers/storage/overlay/d6d3258f4ef8073215491ce035213f00f1e19c3d52c64805e7eabb7580f52ed9/diff:/home/podman/.local/share/containers/storage/overlay/a217ca4a041b2e3c802d3bc552834ad5828f52ea1153df185628f9d8e36bbfb0/diff:/home/podman/.local/share/containers/storage/overlay/1c68fa43e22f7a7523ab8c31cb423254f85d1c28f466ea2402bb8fbce4106eb5/diff:/home/podman/.local/share/containers/storage/overlay/1a23a7e64588a2a119aae6cb178c46a17a1d73e203e551f64944d3ea8a3b1f87/diff:/home/podman/.local/share/containers/storage/overlay/2cfad93a1a539b4e25fc9df0124f8ceb8230056c24afaaf72d091bd05c35490f/diff:/home/podman/.local/share/containers/storage/overlay/1bffdef1381565787bc78f2a991706b7a6a20759a5930b33c4d642c942dde3d0/diff:/home/podman/.local/share/containers/storage/overlay/2d87dd76eed4345a93385b38da441f88c7d64ce20843ca44f81a84bb6e9ddb4a/diff:/home/podman/.local/share/containers/storage/overlay/99d613d1d3d4956138bf0f93ceb14bcef75fe64b75c48c7c7f4babf866fc6a11/diff:/home/podman/.local/share/containers/storage/overlay/9a9d9be17edf85a7680e5fdbea1073e88bcaee2b23d757a1a1732a844418ad2b/diff:/home/podman/.local/share/containers/storage/overlay/54c860da36bc44b63d5504cb72a4b415971f93ef6bcdf9284734b0813867c322/diff:/home/podman/.local/share/containers/storage/overlay/dca32b638ec999e9f0541c06fd396acfb2d27f2b846fd2e3103ee4856e9ec5e6/diff:/home/podman/.local/share/containers/storage/overlay/04135ef8c7586e4eb8d0fb1c5db779983bf43a9ec98d65f3551fdc2c245611aa/diff:/home/podman/.local/share/containers/storage/overlay/4983a1b12f055551c4847ca1352f3a85c8341af3ba65dd60a5e22f3962289e84/diff:/home/podman/.local/share/containers/storage/overlay/cc7d7bab534daca5cd1d486a6b25fdee9e830df4b8f1124e50cc8c6bb5f00714/diff:/home/podman/.local/share/containers/storage/overlay/7b66672bca79fe0166f650b031b8b98480e170210e750a49b47fa496edfc06b3/diff:/home/podman/.local/share/containers/storage/overlay/5d36748e965e25d20fdd8f92309f507193cb4bf0bfe20e0863b9fde7e16a352e/diff:/home/podman/.local/share/containers/storage/overlay/97f42b7f1954b2a9540310cb5369c92cfdc3f6cfcf8667731c36b7fca0713bc0/diff:/home/podman/.local/share/containers/storage/overlay/e609b3e95360c592276b4ab52d6eb2d11ddacfd64e2c77a718a945e0880b4499/diff:/home/podman/.local/share/containers/storage/overlay/ea4d773ba6d0f1e67e07afba32cbe89b7173fe8322ce391f66c59f744e2846fb/diff:/home/podman/.local/share/containers/storage/overlay/f2158a1f2c6e32b5486c1f0f905765fd30e6df757fc473407001f3602bfaa596/diff:/home/podman/.local/share/containers/storage/overlay/487ebb867f71773545105ad390bfa025d6b7eec5d4d2ae846ef57baee4eb4743/diff:/home/podman/.local/share/containers/storage/overlay/22ce6882a446d8608a2f051621b8069fad7bfc45aa7bb8581f394bb97163b154/diff:/home/podman/.local/share/containers/storage/overlay/fce73768ae3669458f9921a17d07dc92bf6a43890e3f8542822cb7cdb7ccf403/diff:/home/podman/.local/share/containers/storage/overlay/b17c615f112ad2c44c8b32e42ddadcea798fe4f0974ab921833d57a15d8ac0c7/diff:/home/podman/.local/share/containers/storage/overlay/2639e165cd0ce4bf9dec4be47011c141b2160d8bbe3e537a93f3aa62c89fdcfb/diff:/home/podman/.local/share/containers/storage/overlay/25a27bb3545ad984cf082233652ac31a89c25b888f5cb20af9ae9ce71243bf45/diff:/home/podman/.local/share/containers/storage/overlay/dc64dab7285b06b312e84cf7c187d7f04a664463488bc353dd69cb67ff49b3f6/diff:/home/podman/.local/share/containers/storage/overlay/19b7b61809165ffd82992c634be20e5f60951ef8f8f1babf6f188b0a248d3587/diff:/home/podman/.local/share/containers/storage/overlay/1b68910348085c416f88f0e988dcfb9701c37dea61a534366c1aef4752472220/diff:/home/podman/.local/share/containers/storage/overlay/35b2f0ed7f6a39fbcba273d6866693c7fc7272848012b83d194411b8881dddf9/diff:/home/podman/.local/share/containers/storage/overlay/8df6858283c1689686720e71ed926a84294eb390f4bfa4f3919863e35e4496d3/diff:/home/podman/.local/share/containers/storage/overlay/181441545c07caa61d6185eb5408b4c0155d0c43d672bbaeb78c9e8e7466faff/diff:/home/podman/.local/share/containers/storage/overlay/b11b98b1718223623a3720a35674a529075923d0d162c16f5d87bac0b8d1f735/diff:/home/podman/.local/share/containers/storage/overlay/c639220f7158b1fe8fcdc44c7392bef2a0c5d01fab3962a0afa793beb49f5f96/diff:/home/podman/.local/share/containers/storage/overlay/4d12c3554df49cddba417c00b1108d7b38f0770d0b1f489a4228fe2ed65bca58/diff:/home/podman/.local/share/containers/storage/overlay/c260951369d53965e25a92b1efe7624b09c79d588b6fe9a8f0dabfd5388ab2cb/diff:/home/podman/.local/share/containers/storage/overlay/a23015ca97b40363dabf0dac86e247901af363e5878b6a7139cdd4094b44c3b4/diff:/home/podman/.local/share/containers/storage/overlay/8e94d61c1e01fa9fd360efd533595c6e2ed1426a1d511f8869597e740ca2f573/diff:/home/podman/.local/share/containers/storage/overlay/e4a39461523976c04e930d56a33b12096c6afc75d8caec96469b679c6e2bc9ba/diff:/home/podman/.local/share/containers/storage/overlay/82629d83413ae787afa14a7eef584403a7aa5d654f121f7d8642c42346bf3fb6/diff:/home/podman/.local/share/containers/storage/overlay/cf50af3ef789b2285f915a223e5c21a009c8354e41ce37159aeb7675db6af486/diff:/home/podman/.local/share/containers/storage/overlay/5065d6a43ab003119f001ac982c6c88a936fb6b643aed2d014e6dc3f29deac08/diff:/home/podman/.local/share/containers/storage/overlay/90edacf9d313dc112c0a1c7387bc42a389dd8c24b2b249d6c0a34869ceb52b6f/diff:/home/podman/.local/share/containers/storage/overlay/202b9cf3e0c249c20bb5672b8f027b218047c3be321423a3a45b6dd5cb940360/diff:/home/podman/.local/share/containers/storage/overlay/a287fda113d42d8defb66724a6bda8a42f2a139c09b13ea7ee10aa5bf0bd706b/diff:/home/podman/.local/share/containers/storage/overlay/d4604cb7d2ad6925345446b0ed0df79bd3540100fa6c3ab98ba187285c3c8a1c/diff:/home/podman/.local/share/containers/storage/overlay/f4f84366778e2c2b75399bc6b38828d29f4d26169b57253e1e910b77c2472e78/diff:/home/podman/.local/share/containers/storage/overlay/9908b8c3e7476b0b4186c8a5e9a5404e3e902d11e0327916b1ba2bfe613e7e46/diff:/home/podman/.local/share/containers/storage/overlay/d7e4656076f6c5ff343ab341289ed0ed65e1482bf61bd729b7a70bab4d9a23d1/diff:/home/podman/.local/share/containers/storage/overlay/f6a1cf1b29b983b1c748218b29ea1c6e9bfc08170d96426e3e51d52f3cf2d90e/diff:/home/podman/.local/share/containers/storage/overlay/d68c172b46feedcbfb83f0df07eb9598f24531ff09be16b9c0d7b7d0fba095ab/diff:/home/podman/.local/share/containers/storage/overlay/267a979d5cffbfb8193b3cc292a0ad52759263e947c172518e8648138ad5285c/diff:/home/podman/.local/share/containers/storage/overlay/2041f432cf668792670347ad2bc3819a6916de929f17b841f185012e50b65015/diff:/home/podman/.local/share/containers/storage/overlay/942e107bcabf09f9e27a80846229adf0dc09e6ab385bd9b84d50c995d558325a/diff:/home/podman/.local/share/containers/storage/overlay/13c357eb68ba105826a8ce2ff3868249b1901e5fe38f8d16e694d21f367df9fe/diff:/home/podman/.local/share/containers/storage/overlay/69a23d25669d569206544e928cd8273285d74e315703e37eaa7458040126219f/diff:/home/podman/.local/share/containers/storage/overlay/e86a1544869c1bbd5c77720fa31601a58135329ec48dc1744c3e3e66667bfb4f/diff:/home/podman/.local/share/containers/storage/overlay/4b9969085ee3a428e933369a313c60dab98b7b28b21acaf13a645c15972ddae7/diff:/home/podman/.local/share/containers/storage/overlay/2fcafe8ff628658acfbf5c5e0f9665afdb0cedb4feba06aab42a84d655a405a0/diff:/home/podman/.local/share/containers/storage/overlay/ae4fa0d6822cd001daecd178d1ff3b8a71f9b1f09c278c6125a64aaec0e50d35/diff:/home/podman/.local/share/containers/storage/overlay/4faee6408455200e239fca199443b40d8cc7aab197ad83a473b6e223a29a7f69/diff:/home/podman/.local/share/containers/storage/overlay/818b5d2b05ac2eae6223b8a51e43af5062ba0dca936fed81d9413355d5f50331/diff:/home/podman/.local/share/containers/storage/overlay/6497ca0b81b636469d7ffe3f573a89d73fd2934c0289d047daeb1aa42ae17dfe/diff:/home/podman/.local/share/containers/storage/overlay/79e86fe45625d85580958819980e94616c582fc3068260ffba03e92c253bbbf9/diff:/home/podman/.local/share/containers/storage/overlay/4c5701839128771b4acceffc234c36e0d85d459269b60958cb9eeeaf63bc193b/diff:/home/podman/.local/share/containers/storage/overlay/0e895cc3077674af6e557e3fc28a9004f95710acd6cdd933e1f1f6bd969cf153/diff:/home/podman/.local/share/containers/storage/overlay/a0b8705a48caae030b90c62f8aadcf65d8f0756fd0a0954b0b1ab67966050af5/diff:/home/podman/.local/share/containers/storage/overlay/cc47a6f63221fb3f4bac3a5d0deb10e0061e49f88355cadd4e696030e6c78e5d/diff:/home/podman/.local/share/containers/storage/overlay/c9966e2d413a876897816cdea8ca3dde1883d085370448c06c31e4d71fc4f44a/diff:/home/podman/.local/share/containers/storage/overlay/54126cd06c7ba016b06afa25f3f5c486aef95945cdd154467f146df305ee4921/diff:/home/podman/.local/share/containers/storage/overlay/7f52531a0e0d94cef849ef310b2be166152f87005e225a4865d95040b4dc0f87/diff:/home/podman/.local/share/containers/storage/overlay/fcadb7e28c681ec6e29ee6a10ea2976f193784d47e906b69d96d153bf6f65db5/diff:/home/podman/.local/share/containers/storage/overlay/455b52c8ba9ab0132e6703307ec377e026e0bed9ecd130a67e5553a8b0d74eda/diff:/home/podman/.local/share/containers/storage/overlay/0c1bb6ee8e43a602a6647ff38ce018cf7951ade514f2a744fff0fa94ab8e83ec/diff:/home/podman/.local/share/containers/storage/overlay/87a9deb6cb215745e690f7bbba8fbab5999bf8ceb4d0608da529e663a19ac976/diff:/home/podman/.local/share/containers/storage/overlay/0d5dcff8c2f3a64e7aa54961fe4077b01c141f22851e808481355844817f2aae/diff:/home/podman/.local/share/containers/storage/overlay/894dc2fdd7b10ba747ea2c463e430783f0b53f8207796d5e2005e31b79a11bdd/diff:/home/podman/.local/share/containers/storage/overlay/59c15d0716e2bcf829fe394b2cd3dc654a1cda2fed23cd62c8ead86e7502b84d/diff:/home/podman/.local/share/containers/storage/overlay/03a934b6a4874638bf5426e9b70ed2909267704b8357f4c37452bffd293351c9/diff:/home/podman/.local/share/containers/storage/overlay/396e7863ae2a885990dfe427a10525f93e0781a68a694abd3734633f6251a056/diff:/home/podman/.local/share/containers/storage/overlay/302a8a090621ebfda8367c22697180b7c3993107b61581bed591626f9110baf4/diff:/home/podman/.local/share/containers/storage/overlay/14b7d92b6928aafdfbb53942dc1c4c3b0aea1d825295fd94dc3191da50b2bfcc/diff:/home/podman/.local/share/containers/storage/overlay/200e7a3cc9c1b0f84aaa78d12e9a0c6cfd636accd6faf37b2e5a26957abd884a/diff:/home/podman/.local/share/containers/storage/overlay/64d55d0be5307b6a8360343b150481f90458ec777d4dc5de6483d6ba6b0f81cd/diff:/home/podman/.local/share/containers/storage/overlay/c8968be30870e88d6cc1b9f8f7a6192ec4b013fa08da48c12bd17571edacc98a/diff:/home/podman/.local/share/containers/storage/overlay/8f6185e425a421fbeaab564fdff4b2b6d32a59d6abf8a43990118e5f47ed8d9d/diff:/home/podman/.local/share/containers/storage/overlay/729cb4973d37eb5f9ef68ecc80f6793baa459e49f471233a433a27bfa85f9f84/diff:/home/podman/.local/share/containers/storage/overlay/687951c2071a6dcfbd7c94444786fc35a75508ff22ca239ae5711cc6eb50e9e1/diff:/home/podman/.local/share/containers/storage/overlay/1dd1c72ab53a2cde698cb05dd5158319d525fa1aee846f005f4297d0cbd8bc44/diff:/home/podman/.local/share/containers/storage/overlay/a14ddf69584a9ef2e6a664698eaea949f957883e29f39e820fddca10267c7cfe/diff:/home/podman/.local/share/containers/storage/overlay/c00af022a6bb72d2ae8c12032c580e24ce26c4d6cc2527ae9bd43d50b2f99de2/diff:/home/podman/.local/share/containers/storage/overlay/55f56a1e345590056c0809e6d60ebd5839056c4f995554f72224cb596d5b8e34/diff:/home/podman/.local/share/containers/storage/overlay/2f308593d4a7320332f910effb544c84f65c35a9a3bc70f13efdc08ee1999cb4/diff:/home/podman/.local/share/containers/storage/overlay/08fc6166d3e62dd12802f7fd42d619dd82225aff0e0ba687f357ea5fdcfccb62/diff:/home/podman/.local/share/containers/storage/overlay/b4348bfd84050f05a39ebe017b137d3bde5ce307cc5dd819495c4f362587c4e3/diff:/home/podman/.local/share/containers/storage/overlay/9dd0b45ebe39e477b2a869802e9b771d991f7341cb79e3446c9ca8231fc35d3f/diff:/home/podman/.local/share/containers/storage/overlay/c95d2191d7773c6e29188f92922bc9547e1f0b6130e85dfc2f5e4eae13137c7c/diff",
                "UpperDir": "/home/podman/.local/share/containers/storage/overlay/df14225875fcefabe1a37efef2256b421df5e60732449f2f87386882a6b7e275/diff",
                "WorkDir": "/home/podman/.local/share/containers/storage/overlay/df14225875fcefabe1a37efef2256b421df5e60732449f2f87386882a6b7e275/work"
            }
        },
        "Mounts": [],
        "Dependencies": [],
        "NetworkSettings": {
            "EndpointID": "",
            "Gateway": "",
            "IPAddress": "",
            "IPPrefixLen": 0,
            "IPv6Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "MacAddress": "",
            "Bridge": "",
            "SandboxID": "",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {
                "80/tcp": [
                    {
                        "HostIp": "",
                        "HostPort": "2080"
                    }
                ]
            },
            "SandboxKey": ""
        },
        "ExitCommand": [
            "/usr/bin/podman",
            "--root",
            "/home/podman/.local/share/containers/storage",
            "--runroot",
            "/run/user/1000/containers",
            "--log-level",
            "error",
            "--cgroup-manager",
            "cgroupfs",
            "--tmpdir",
            "/run/user/1000/libpod/tmp",
            "--runtime",
            "crun",
            "--storage-driver",
            "overlay",
            "--storage-opt",
            "overlay.mount_program=/usr/bin/fuse-overlayfs",
            "--events-backend",
            "journald",
            "container",
            "cleanup",
            "982b3d16588567bd5a7901c875a12cb52f5bc1537081cc9a98786bfb3e2d0925"
        ],
        "Namespace": "",
        "IsInfra": false,
        "Config": {
            "Hostname": "982b3d165885",
            "Domainname": "",
            "User": "dockware",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "TERM=xterm",
                "container=podman",
                "SW_CURRENCY=not-set",
                "XDEBUG_ENABLED=0",
                "PHP_IDE_CONFIG=serverName=localhost",
                "MYSQL_USER=not-set",
                "TIDEWAYS_KEY=not-set",
                "TIDEWAYS_SERVICE=web",
                "XDEBUG_REMOTE_HOST=host.docker.internal",
                "FILEBEAT_ENABLED=0",
                "TIDEWAYS_ENV=production",
                "PHP_VERSION=7.4",
                "MYSQL_PWD=not-set",
                "SSH_PWD=not-set",
                "SSH_USER=not-set",
                "TZ=Europe/Berlin",
                "containers=podman",
                "APACHE_DOCROOT=/var/www/html/public",
                "XDEBUG_CONFIG=idekey=PHPSTORM",
                "HOSTNAME=982b3d165885",
                "HOME=/var/www"
            ],
            "Cmd": [
                "/entrypoint.sh"
            ],
            "Image": "docker.io/dockware/dev:latest",
            "Volumes": null,
            "WorkingDir": "/var/www/html",
            "Entrypoint": "",
            "OnBuild": null,
            "Labels": {
                "maintainer": "dasistweb GmbH",
                "title": "Dockware dev:6.3.5.0",
                "version": "1.3.5-beta"
            },
            "Annotations": {
                "io.container.manager": "libpod",
                "io.kubernetes.cri-o.Created": "2021-02-08T16:25:54.8045019+01:00",
                "io.kubernetes.cri-o.TTY": "false",
                "io.podman.annotations.autoremove": "FALSE",
                "io.podman.annotations.init": "FALSE",
                "io.podman.annotations.privileged": "FALSE",
                "io.podman.annotations.publish-all": "FALSE",
                "org.opencontainers.image.stopSignal": "15"
            },
            "StopSignal": 15,
            "CreateCommand": [
                "podman",
                "run",
                "-p",
                "2080:80",
                "dockware/dev:latest"
            ],
            "Umask": "0022"
        },
        "HostConfig": {
            "Binds": [],
            "CgroupMode": "private",
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "k8s-file",
                "Config": null
            },
            "NetworkMode": "slirp4netns",
            "PortBindings": {
                "80/tcp": [
                    {
                        "HostIp": "",
                        "HostPort": "2080"
                    }
                ]
            },
            "RestartPolicy": {
                "Name": "",
                "MaximumRetryCount": 0
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": null,
            "CapAdd": [],
            "CapDrop": [],
            "Dns": [],
            "DnsOptions": [],
            "DnsSearch": [],
            "ExtraHosts": [],
            "GroupAdd": [],
            "IpcMode": "private",
            "Cgroup": "",
            "Cgroups": "default",
            "Links": null,
            "OomScoreAdj": 0,
            "PidMode": "private",
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": [],
            "Tmpfs": {},
            "UTSMode": "private",
            "UsernsMode": "",
            "ShmSize": 65536000,
            "Runtime": "oci",
            "ConsoleSize": [
                0,
                0
            ],
            "Isolation": "",
            "CpuShares": 0,
            "Memory": 0,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": null,
            "BlkioDeviceReadBps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteIOps": null,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": [],
            "DiskQuota": 0,
            "KernelMemory": 0,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": 0,
            "OomKillDisable": false,
            "PidsLimit": 0,
            "Ulimits": [
                {
                    "Name": "RLIMIT_NOFILE",
                    "Soft": 1048576,
                    "Hard": 1048576
                },
                {
                    "Name": "RLIMIT_NPROC",
                    "Soft": 32768,
                    "Hard": 32768
                }
            ],
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0,
            "CgroupConf": null
        }
    }
]

podman@cloud2-vm400:~$
```

I edited the `entrypoint.sh` to get this information:

```
id:
uid=33(www-data) gid=33(www-data) groups=33(www-data)
root@dd8dd540f6fe:~# cat /etc/passwd | grep -E "www-data|dockware"
www-data:x:33:33:www-data:/var/www:/usr/sbin/nologin
dockware:x:33:33:,,,:/var/www:/bin/bash
root@dd8dd540f6fe:~# 
```

It seems like the image looks correct, but the uid is duplicated. Is this correctly?

**Comment by boxblinkracer** [comment]
Hi

it looks a bit different for me,
can you add this to the entrypoint.sh

`id`
`whoami`

this is my output that shows also the sudo.
i think thats the problem, its missing in your setup

`uid=33(www-data) gid=33(www-data) groups=33(www-data),27(sudo)`
`www-data`

**Comment by lroehrs** [comment]
Hi,

here:

```
podman@cloud2-vm400:~$ podman start -a fc2ff9a4b8d8
whoami:
www-data
id:
uid=33(www-data) gid=33(www-data) groups=33(www-data)
podman@cloud2-vm400:~$ 
```

In my is the sudo missing. Do you have an idea why?

**Comment by lroehrs** [comment]
@boxblinkracer ping..

**Comment by boxblinkracer** [comment]
Hi
sorry for the delay
are you in the shopware slack channel? can you write me there, then i would love to do a zoom with you if possible

**Comment by boxblinkracer** [comment]
to everyone who's watching - we had a zoom call. somehow the plain podman launch does not work, it seems as if the host www-data is somehow being used without sudo permission.
actually the used dockware user is only an alias for www-data (33), but it will accidentally show www-data which shouldnt make a problem, though it kinda gets switched to a real host www-data in podman somehow

he will investiage some ideas and come back to me

**Comment by boxblinkracer** [comment]
issue was resolved by requester with the new boot_start script implementation of dockware.
reason was that podman somehow reuses the original www-data of the host which comes without sudo obviously.

the dockware www-data comes also without sudo, but the actual user being used is the 'dockware' user, which is just an alias of www-data UID 33.
we cant change that for now, but for podman its possible to add this line in the boot start script

`sudo sed -i 's/dockware:x:33:33:/dockware:x:7788:33:/g' /etc/passwd` 

this changes the dockware uid to 7788, so its then not accidentally recognized as www-data and does does really use the dockware user when booting which has the correct sudo NOPWD access

**Comment by af-ans** [comment]
Hello, sorry for reopening such an old issue, but we are currently facing the same issues on a rootless podman setup.
I used the dockware/play image for any tests. 
The Fix with the boot script don't seem to work (anymore), even when supplying the new UID (7788) as user.
Additionally, we need to use the keep-groups flag for adding the needed supplementary groups.
I only got the container to run using root as the user, which leads me to the questing why the dockware user is needed in the first place, when it has sudo without restrictions anyways.

Thank you for any help!
